test/onnx/dynamo/test_registry_dispatcher.py

```diff
@@ -191,304 +87,337 @@ def setUp(self):
-        )
-        self.assertEqual(
-            onnxfunction_dispatcher._find_onnx_data_type(
-                torch.tensor([1], dtype=torch.float16)
+            ([torch.randn(3, 4), torch.randn(3, 4)], {}, ops.core.aten_mul_bool, 0),
+            (
+                [
+                    torch.randint(0, 2, size=(3, 4), dtype=torch.int).bool(),
+                    torch.randint(0, 2, size=(3, 4), dtype=torch.int).bool(),
+                ],
+                {},
+                ops.core.aten_mul_bool,
+                2,
             ),
-            {"tensor(float16)"},
-        )
-        self.assertEqual(
-            onnxfunction_dispatcher._find_onnx_data_type(
-                torch.tensor([1], dtype=torch.int32)
+        ]
+    )
+    def test_matching_score_system_on_overload_dtypes(self, inputs, kwargs, op, score):
+        op_schema_wrapper = onnxfunction_dispatcher._OpSchemaWrapper(op.op_schema)
+        op_schema_wrapper._record_matching_score(inputs, kwargs)
+        self.assertEqual(op_schema_wrapper.match_score, score)
+
+    @parameterized.expand(
+        [
+            ([torch.randn(3, 4), torch.tensor(3)], {}, ops.core.aten_new_full, 2),
+            (
+                [torch.randn(3, 4), torch.tensor(3)],
+                {"dtype": torch.float},
+                ops.core.aten_new_full,
+                1,
             ),
-            {"tensor(int32)"},
-        )
-        self.assertEqual(
-            onnxfunction_dispatcher._find_onnx_data_type(
-                [torch.tensor([1], dtype=torch.int32)]
+            ([torch.randn(3, 4), torch.tensor(3)], {}, ops.core.aten_new_full_dtype, 1),
+            (
+                [torch.randn(3, 4), torch.tensor(3)],
+                {"dtype": torch.float},
+                ops.core.aten_new_full_dtype,
+                2,
             ),
-            {"seq(tensor(int32))"},
+        ]
+    )
+    def test_matching_score_system_on_optional_dtypes(self, inputs, kwargs, op, score):
+        op_schema_wrapper = onnxfunction_dispatcher._OpSchemaWrapper(op.op_schema)
+        op_schema_wrapper._record_matching_score(inputs, kwargs)
+        self.assertEqual(op_schema_wrapper.match_score, score)
+
+    @parameterized.expand(
+        [
+            (1, {"tensor(int64)", "tensor(int16)", "tensor(int32)"}),
+            (1.0, {"tensor(float)", "tensor(double)", "tensor(float16)"}),
+            (torch.tensor([True]), {"tensor(bool)"}),
+            (torch.tensor([1], dtype=torch.int64), {"tensor(int64)"}),
+            (torch.tensor([1], dtype=torch.int32), {"tensor(int32)"}),
+            (torch.tensor([1], dtype=torch.int16), {"tensor(int16)"}),
+            (torch.tensor([1], dtype=torch.float), {"tensor(float)"}),
+            (torch.tensor([1], dtype=torch.float16), {"tensor(float16)"}),
+            (torch.tensor([1], dtype=torch.double), {"tensor(double)"}),
+            (None, set()),  # None is not type allowed
+            ([], set()),  # Empty list is not type allowed
```

Comment:
What does "not allowed" mean? I would clarify

Reply:
rephrased